### ISSUE TITLE
Crash in dalmatiner_read_fsm when trying to read repair

### ISSUE THREAD

**Issue by starstuck** [outlyerapp/dalmatinerdb#12]
Exception exit:

``````
{{function_clause,[{riak_core_vnode_master,command2 [730750818665451459101842416358141509827966271488,{repair,<<"dalmatinerdb">>,<<24,100,97,
108,109,97,116,105,110,101,114,100,98,64,49,55,50,46,49,54,46,49,48,46,49,3,103,101,116,3,109,97,120>>,1455196518,<<1,0,0,0,0,12,197,246,1,0,0,0,0,12,197,246,1,0
,0,0,0,12,197,246,1,0,0,0,0,12,197,246,1,0,0,0,0,12,197,246,1,0,0,0,0,12,197,246,1,0,0,0,0,12,197,246,1,0,0,0,0,12,197,246,1,0,0,0,0,12,197,246,1,0,0,0,0,12,197,
246,1,0,0,0,0,12,197,246,1,0,0,0,0,12,197,246,1,0,0,0,0,12,197,246,1,0,0,0,0,12,197,246,1,0,0,0,0,12,197,246,1,0,0,0,0,12,197,246,1,0,0,0,0,12,197,246,1,0,0,0,0,
12,140,132,1,0,0,0,0,12,140,132,1,0,0,0,0,12,140,132,1,0,0,0,0,12,140,132,1,0,0,0,0,12,140,132,1,0,0,0,0,12,140,132,1,0,0,0,0,12,140,132,1,0,0,0,0,12,140,132,1,0
,0,0,0,12,140,132,1,0,0,0,0,12,140,132,1,0,0,0,0,12,140,132,1,0,0,0,0,10,173,83,1,0,0,0,0,10,173,83>>},ignore,metric_vnode_master,normal],[{file,"/data/jenkins/w
orkspace/dalmatinerdb/_build/default/lib/riak_core/src/riak_core_vnode_master.erl"},{line,81}]},{dalmatiner_read_fsm,repair,4,[{file,"/data/jenkins/workspace/dal
matinerdb/_build/deb/lib/dalmatiner_db/src/dalmatiner_read_fsm.erl"},{line,270}]},{dalmatiner_read_fsm,finalize,2,[{file,"/data/jenkins/workspace/dalmatinerdb/_b
uild/deb/lib/dalmatiner_db/src/dalmatiner_read_fsm.erl"},{line,195}]},{gen_fsm,handle_msg,7,[{file,"gen_fsm.erl"},{line,518}]},{proc_lib,init_p_do_apply,3,[{file
,"proc_lib.erl"},{line,240}]}]},[{gen_fsm,terminate,7,[{file,"gen_fsm.erl"},{line,626}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]}
    ancestors: [dalmatiner_read_fsm_sup,dalmatiner_db_sup,<0.572.0>]```


<!---
@huboard:{"custom_state":"archived"}
-->
``````

**Comment by davecromberge** [comment]
Proposed fix is #11

**Comment by davecromberge** [comment]
@szarsti I think this one is fixed.
